x-pack/plugins/lists/server/services/exception_lists/exception_list_client.ts

```diff
@@ -865,4 +904,257 @@ export class ExceptionListClient {
+   * NOTE: This is slightly different from the saved objects version in that it takes
+   * an injected function, so that we avoid doing additional plumbing with generators
+   * to try to keep the maintenance of this machinery simpler for now.
+   *
+   * If you want to stream all results up to 10k into memory for correlation this would be:
+   * @example
+   * ```ts
+   * const exceptionList: ExceptionListItemSchema[] = [];
+   * const executeFunctionOnStream = (response: FoundExceptionListItemSchema) => {
+   *   exceptionList = [...exceptionList, ...response.data];
+   * }
+   * await client.findExceptionListItemPointInTimeFinder({
+   *   filter,
+   *   executeFunctionOnStream,
+   *   namespaceType,
+   *   maxSize: undefined, // NOTE: This is unbounded when it is "undefined"
+   *   perPage: 1_000,
+   *   sortField,
+   *   sortOrder,
+   *   exe
+   * });
+   * ```
+   * @param filter {string} Your filter
+   * @param namespaceType {string} "agnostic" | "single" of your namespace
+   * @param perPage {number} The number of items per page. Typical value should be 1_000 here. Never go above 10_000
+   * @param maxSize {number of undefined} If given a max size, this will not exceeded. Otherwise if undefined is passed down, all records will be processed.
+   * @param sortField {string} String of the field to sort against
+   * @param sortOrder "asc" | "desc" The order to sort against
+   */
+  public findExceptionListItemPointInTimeFinder = async ({
+    executeFunctionOnStream,
+    filter,
+    listId,
+    maxSize,
+    namespaceType,
+    perPage,
+    sortField,
+    sortOrder,
+  }: FindExceptionListItemPointInTimeFinderOptions): Promise<void> => {
+    const { savedObjectsClient } = this;
+    return findExceptionListItemPointInTimeFinder({
+      executeFunctionOnStream,
+      filter,
+      listId,
+      maxSize,
+      namespaceType,
+      perPage,
+      savedObjectsClient,
+      sortField,
+      sortOrder,
+    });
+  };
+
+  /**
+   * Finds an exception list within a point in time (PIT) and then calls the function
+   * `executeFunctionOnStream` until the maxPerPage is reached and stops.
+   * NOTE: This is slightly different from the saved objects version in that it takes
+   * an injected function, so that we avoid doing additional plumbing with generators
+   * to try to keep the maintenance of this machinery simpler for now.
+   *
+   * If you want to stream all results up to 10k into memory for correlation this would be:
+   * @example
+   * ```ts
+   * const exceptionList: ExceptionListSchema[] = [];
+   * const executeFunctionOnStream = (response: FoundExceptionListSchema) => {
+   *   exceptionList = [...exceptionList, ...response.data];
+   * }
+   * await client.findExceptionListPointInTimeFinder({
+   *   filter,
+   *   executeFunctionOnStream,
+   *   namespaceType,
+   *   maxSize: undefined, // NOTE: This is unbounded when it is "undefined"
```

Comment:
Like one above, think this should be 10,000 for sake of described example.

Reply:
Ok, I think I know what you're putting down here... I shall modify it here and in the other examples.